src/hooks/useYoutubeData.js

```diff
@@ -6,25 +6,25 @@ const { searchListByQuery } = youtube;
 
 export default function useYoutubeData() {
```

Comment:
We could make this hook more flexible by passing `searchListByQuery` as a parameter:

```js
export default function useRequest(request, queries) {
  const [data, setData] = useState(null);
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState(null);

  const fetchDate = useCallback(async () => {
    setIsLoading(true);

    const res = await request(queries);
    try {
        setData(await res.json());
      } catch (e) {
        setError(e.message);
      }

    setIsLoading(false);
  });

  useEffect(() => {
    fetchData();
  }, [fetchData]);

  return { data, isLoading, error };
}

```

Reply:
Hi @qtd-wizeline , I updated it to make use of custom hooks